### ISSUE TITLE
Add speculative decoding workflow for 2-5× throughput

### ISSUE THREAD

**Issue by christopherthompson81** [christopherthompson81/LlamaCpp.Bindings#4]
## Goal

Use a small, fast draft model to predict several tokens ahead, then verify those tokens with the main model in one parallel decode step. Typical speedups of 2–5× on large models when the draft distribution matches the main's closely.

## Native primitives — already bound

All the ingredients are in place:

- `llama_batch_*` supports multiple sequences (we have the struct mirror + `llama_batch_get_one`)
- `llama_memory_seq_cp` branches a sequence
- `llama_sampler_sample` / `llama_sampler_accept`
- `llama_get_logits_ith` returns logits for a specific batch position

What's missing is a public API that orchestrates them into the rejection-sampling protocol.

## Public API shape (proposal)

```csharp
public sealed class LlamaSpeculativeGenerator
{
    public LlamaSpeculativeGenerator(
        LlamaContext main,
        LlamaContext draft,           // second context holding a smaller draft model
        LlamaSampler mainSampler,
        LlamaSampler draftSampler,
        int draftLookahead = 5);

    public IAsyncEnumerable<string> GenerateAsync(
        string prompt, int maxTokens,
        CancellationToken cancellationToken = default);
}
```

## Action items

- [ ] Implement the standard speculative-decoding protocol: draft runs N steps; main verifies N+1 positions in one decode; accept the longest agreeing prefix; roll back the draft's KV beyond the divergence point using `RemoveSequenceRange`
- [ ] Decide whether main and draft share one `LlamaModel` (same weights, different samplers — less useful) or require two distinct models (standard case)
- [ ] Probability scoring for rejection sampling: expose enough of the logits surface to compare draft token prob vs. main token prob at each position
- [ ] Add tests with two small models (e.g., `TinyLlama-1.1B` as draft + a `3B` main) — will need a second model fixture
- [ ] Benchmark: measure tokens/sec speedup on the golden Qwen3 model with a suitable draft
- [ ] Docs + XML comments covering when speculative decoding helps (aligned distributions) vs. hurts (mismatched distributions → many rejections)

## Considerations

- Requires two models loaded simultaneously → 1.5–2× VRAM pressure vs. single-model generation.
- Only wins when the draft agrees with the main most of the time. A draft fine-tuned on a very different corpus will lose to plain single-model decoding.
- CI can't test this without two GGUFs available locally; gate behind an env var similar to the existing `LLAMACPP_TEST_MODEL`.
- Cancellation semantics: a mid-decode cancel needs to roll back BOTH contexts' KV state to a consistent point, or mark them as dirty and require `ClearKvCache` before next use.
